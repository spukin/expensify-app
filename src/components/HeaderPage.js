import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const HeaderPage = ({ startLogout }) => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <Link className="header__title" to="/dashboard" activeClassName="is-active">
                    <h1>Expensify</h1>
                </Link>
                <button className="button button--link" onClick={startLogout}>Log out</button>
            </div>
        </div>
    </header>
)

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(HeaderPage);