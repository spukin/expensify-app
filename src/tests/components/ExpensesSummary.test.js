import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should correctly render expenses summary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={122} />)
    expect(wrapper).toMatchSnapshot();
});

test('Should correctly render expenses summary with more than 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={5} expensesTotal={45343432} />)
    expect(wrapper).toMatchSnapshot();
});