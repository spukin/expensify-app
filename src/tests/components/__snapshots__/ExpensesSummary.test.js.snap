// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render expenses summary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      There are 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $1.22 
      </span>
       ammount
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should correctly render expenses summary with more than 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      There are 
      <span>
        5
      </span>
       
      expenses
       totalling 
      <span>
        $453,434.32 
      </span>
       ammount
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
